File: TFG/ftp_tfg/lib/domain/usecases/save_profile.dart
import '../entities/ftp_profile.dart';
import '../repositories/ftp_repository.dart';
import '../interfaces/i_save_profile_use_case.dart';

class SaveProfile implements ISaveProfileUseCase {
  final FtpRepository repository;

  SaveProfile(this.repository);

  @override
  Future<int> execute(FtpProfile profile, String ownerId) =>
      repository.saveProfile(profile, ownerId);
}

File: TFG/ftp_tfg/hotftp_api/src/application/use-cases/profiles/save-profile.ts
import type { FtpProfile } from '../../../domain/entities/ftp-profile.js';
import type { ProfileRepository } from '../../../domain/repositories/profile-repository.js';

export class SaveProfile {
  constructor(private readonly profileRepository: ProfileRepository) {}

  execute(profile: FtpProfile): Promise<FtpProfile> {
    return this.profileRepository.save(profile);
  }
}

